src/NServiceBus.NHibernate/Outbox/OutboxCleaner.cs

```diff
@@ -21,7 +23,7 @@ protected override Task OnStart(IMessageSession busSession)
 
             if (configValue == null)
             {
-                timeToKeepDeduplicationData = TimeSpan.FromDays(7);
+                timeToKeepDeduplicationData = settings.GetOrDefault<TimeSpan?>("Outbox.TimeToKeepDeduplicationData") ?? TimeSpan.FromDays(7);
```

Comment:
Agreed, in general I believe the idea it to "Not use settings in the runtime code". Ie in the features translate settings into explicit args passed to pipeline stuff, startuptasks etc